### DISCUSSION TITLE
OCR-Error after Update pikedpf 8.12.0 to 8.13.0 for a few documents

### DISCUSSION THREAD

**Discussion by afrank73** [ocrmypdf/OCRmyPDF#1288] [Q&A] [answered]
We use paperless-ngx version 2.6.3 to archive our documents. With this current version, errors are shown in the log for a few digital invoices, which we did not have with a previous version. We suspect that this is due to a newer version of "pikepdf", with which "ocrmypdf" may be more restrictive. We did not have this error with version 8.12.0. The error occurred for the first time with version 8.13.0. The version of "ocrmypdf" is unchanged at 15.4.4.
We were able to determine via the AdobePro software that it is a QR image that is causing this error.

The error log from paperless-ngx is as follows:

[2024-03-29 19:48:25,125] [INFO] [ocrmypdf._pipelines.ocr] Postprocessing...
[2024-03-29 19:48:25,462] [ERROR] [ocrmypdf.optimize] xref 41: While extracting this image, an error occurred
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/optimize.py", line 323, in extract_images
    result = extract_fn(
             ^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/ocrmypdf/optimize.py", line 162, in extract_image_jbig2
    ext = pim.extract_to(stream=f)
          ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pikepdf/models/image.py", line 709, in extract_to
    return self._extract_to_stream(stream=stream)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pikepdf/models/image.py", line 649, in _extract_to_stream
    direct_extraction = self._extract_direct(stream=stream)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pikepdf/models/image.py", line 541, in _extract_direct
    stream.write(self._generate_ccitt_header(data, icc=icc))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pikepdf/models/image.py", line 776, in _generate_ccitt_header
    decode = self._decode_array
             ^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pikepdf/models/image.py", line 181, in _decode_array
    raise NotImplementedError(
NotImplementedError: Don't how to retrieve default /Decode array for image<pikepdf.PdfImage image mode=? size=200x198 at 0x7f52f62cf610

I would like to make this invoice available to the maintainer, but I don't know how.
Could someone help me?

**Comment by jbarlow83** [comment, marked as answer]
The only thing changed from pikepdf 8.12 to 8.13 releasing more binary wheels for PyPy and tweak a test that randomly failed because it took long. You're not using PyPy or running the test suite, so it had to be something else that changed because 8.12 and 8.13 are the same code.

This type of error is also not fatal. It only means we can't optimize the image because it violated some expectations about images we can optimize.

You can GPG encrypt your file with my public key as described here:
https://github.com/ocrmypdf/OCRmyPDF/wiki
Alternately you can email me. My email is listed on this page
https://pypi.org/project/pikepdf/

**Reply by afrank73** [reply]
Thank you very much for the quick reply.

I have sent you the invoice by e-mail

**Comment by jbarlow83** [comment]
I could not reproduce the issue with the latest ocrmypdf and pikepdf. 

The provided PDF (note: saved as 1288.pdf) contains errors/was incorrectly generated.

`qpdf --check` reports:
```
checking issuepdf/1288.pdf
PDF Version: 1.6
File is not encrypted
File is linearized
WARNING: linearized file contains an uncompressed object after a compressed one in a cross-reference stream
WARNING: part 8 is empty but nshared_total > nshared_first_page
WARNING: object count mismatch for page 0: hint table = 22; computed = 21
WARNING: page 0 has shared identifier entries
WARNING: page 0: shared object 28: in hint table but not computed list
WARNING: object count mismatch for page 1: hint table = 5; computed = 6
WARNING: page 1: shared object 6: in hint table but not computed list
WARNING: page 1: shared object 29: in hint table but not computed list
qpdf: operation succeeded with warnings
```

`pdfimages -list` reports:
```
page   num  type   width height color comp bpc  enc interp  object ID x-ppi y-ppi size ratio
--------------------------------------------------------------------------------------------
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: Could not parse ligature component "logo" of "t_logo_rgb_k" in parseCharName
Syntax Warning: Could not parse ligature component "rgb" of "t_logo_rgb_k" in parseCharName
   1     0 stencil   200   198  -       1   1  ccitt  no        49  0   300   300  806B  16%
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
Syntax Warning: FoFiType1::parse a line has more than 255 characters, we don't support this
```

**Reply by afrank73** [reply]
Since it's not a fatal error and the documents are still read and recognized in paperless-ngx, I'm fine with it.
If there are further problems, I will open a new thread.
Thank so much.
